src/TSTransformer/nodes/class/transformDecorators.ts

```diff
@@ -141,22 +141,23 @@ function transformParameterDecorators(
 ): luau.List<luau.Statement> {
 	const result = luau.list.make<luau.Statement>();
 
+	const memberName = member.name;
+	if (!memberName || !ts.isIdentifier(memberName) || ts.isPrivateIdentifier(memberName)) return result;
+	const memberNameText = memberName.text;
```

Comment:
This isn't entirely correct, it doesn't support computed properties, string literals or numeric literals. Plain member decorators do support this, so you can take a look at how the member decorators retrieve the name.

Reply:
To be clear, do member decorators retrieve their name with the following code?:
https://github.com/roblox-ts/roblox-ts/blob/81ffb739a133db1ac39edc759f0ad2c517366058/src/TSTransformer/nodes/class/transformDecorators.ts#L48

Reply:
Yes. The first `key = ` will check if the member itself decided to generate a prereq statement. You'll likely need to update `transformMethodDeclaration` to generate the prereq when a parameter is decorated, I believe it'll only do so if there is a member decorator currently.

The if condition will just transform the key as-is, if the member didn't create a prereq (meaning its key is "simple" and won't cause side effects if referenced multiple times in code)

Reply:
I think this should be working now, but I'm not sure how to create a method where the prereq statement would be generated, so I don't know of a way to test it.

Reply:
```ts
let a = "a";
let b = "b";
let cond = true as boolean;

class Test {
    [cond ? a : b](@decorate param: string): void;
}
```